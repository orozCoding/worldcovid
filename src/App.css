@import url('https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;0,700;0,900;1,100;1,300;1,400;1,700;1,900&display=swap');
@import url("https://cdn.jsdelivr.net/npm/bootstrap-icons@1.8.1/font/bootstrap-icons.css");

/* global classes */ 

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.lato {
  font-family: 'Lato', sans-serif;;
}

.bold {
  font-weight: bold;
}

.d-flex {
  display: flex;
  justify-content:  center;
  align-items: center;
}

.col {
  flex-direction: column;
}

.ck {
  cursor: pointer;
}

html {
  text-align: center;
  width: 100%;
  background-color: white;
}

body {
  max-width: 500px;
  background-color: grey;
  color: white;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
  overflow: scroll;
  text-align: center;
  margin:auto;
}

#root {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%
}

.main {
  width: 100%;
  background-color: #4369b2;
  min-height: 97vh;
}

/* end of global classes */ 

.home-left,
.home-right {
  min-width: 50%;
  padding: 50px 0;
  height: 150px;
}

.home-left {
  background-color: #3d62a7;
}

.mainTitle {
  font-size: 30px;
  font-weight: bold;
}

.subTitle {
  width: 100%;
  font-size: 20px;
}

.separator {
  padding: 10px 0;
  border-top: solid 2px #3b5ea0;
  border-bottom: solid 2px #3b5ea0;
}

.totalTitle {
  font-size: 18px;
}

.countriesContainer {
  flex-wrap: wrap;
}

.countryBox {
  width: 50%;
  height: 150px;
}

.countryBox:hover {
  background-color: #4870bb;
}

.countryBox:nth-child(4n-1),
.countryBox:nth-child(4n-2) {
  background-color: #3d62a7;
}

.countryBox:nth-child(4n-1):hover,
.countryBox:nth-child(4n-2):hover {
  background-color: #385996;
}

.arrowIcon {
  height: 20px;
  padding: 15px 10px;
  width: 100%;
  align-self: flex-end;
  justify-content: end;
}

.arrowIcon i {
  margin-left: auto;
}

.countryInfo {
  width: 100%;
  margin-top: -35px;
  height: 150px;
  text-align: right;
}

.countryInfo div {
  margin-left: auto;
  padding-right: 10px;
}

.countryName {
  font-size: 25px;
}

/* single country style */

.selfCountries {
  width: 100%;
  justify-content: start;
  align-items: flex-start;
  margin-bottom: auto;
}

.countryContainer {
  width: 100%;
  flex-grow: 1;
}

.selfTitle {
  width: 100%;
  font-size: 60px;
  min-height: 130px;
  justify-content: end;
  padding: 20px;
}


.selfDate {
  width: 100%;
  gap: 30px;
  background-color: #3b5ea0;
  padding: 10px 0;
  font-size: 20px;
}

.selfStats {
  width: 100%;
}

.selfStats div {
  font-size: 20px;
  width: 100%;
  padding: 10px 80px;
  justify-content: space-between;
}

.selfStats div:nth-child(even) {
  background-color: #3b5ea0;
}

.backButton {
  padding: 30px;
  font-size: 35px;
}