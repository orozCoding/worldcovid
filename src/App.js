import React, { useEffect } from 'react';
import './App.css';
import WorldHome from './components/world/worldHome';
import { BrowserRouter as Router, Routes, Route, Navigate } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { loadCountriesData } from './redux/countries/countries';
import { loadWorldData } from './redux/world/world';
import SingleCountry from './components/countries/singleCountry';
import filterName from './components/filterName';

const App = () => {
  const countriesData = useSelector((state) => state.countries);
  const worldData = useSelector((state) => state.world);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!countriesData.length) {
      dispatch(loadCountriesData());
    }
  }, []);

  useEffect(() => {
    if (!worldData.length) {
      dispatch(loadWorldData());
    }
  }, []);


  return (
    <Router>
      <main className="main d-flex col">
        <Routes>
          <Route path="/" element={<WorldHome worldData={worldData} countriesData={countriesData} />} />
          {
            countriesData.map((country) => {
              return <Route key={country.id} path={
                `/${filterName(country.name)}`
              } element={<SingleCountry key={country.id} country={country} />} />
            })
          }
          <Route
            path="*"
            element={<Navigate to="/" />}
          />
        </Routes>
      </main>
    </Router>
  );
}

export default App;
