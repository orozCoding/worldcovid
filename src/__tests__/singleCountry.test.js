import React from 'react';
import {
  render, screen, fireEvent, wait,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router, Routes, Route, Navigate,
} from 'react-router-dom';
import App from '../App';
import store from '../redux/configStore';
import worldInfo, { albania } from '../components/testsComponents/statsMock';
import SingleCountry from '../components/countries/singleCountry';

let mockData;

beforeEach(() => {
  jest.spyOn(global, 'fetch').mockResolvedValue({
    json: jest.fn().mockResolvedValue(mockData),
  });
});

afterEach(() => {
  jest.restoreAllMocks();
});

describe('Single Country', () => {
  mockData = worldInfo;
  it('renders', () => {
    const tree = renderer
      .create(
        <Router><SingleCountry country={albania} /></Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays country name', () => {
    mockData = worldInfo;
    render(<Router><SingleCountry country={albania} /></Router>);
    expect(screen.getByText('Albania')).toBeDefined();
  });

  it('displays country stats', () => {
    mockData = worldInfo;
    render(<Router><SingleCountry country={albania} /></Router>);
    expect(screen.getByText('Deaths today:')).toBeDefined();
  });
});
