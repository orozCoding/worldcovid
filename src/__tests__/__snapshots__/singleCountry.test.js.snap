// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Countries renders 1`] = `
<div
  className="selfCountries d-flex col"
>
  <div
    className="countryContainer d-flex col"
  >
    <div
      className="selfTitle lato bold d-flex"
    >
      <p>
        Albania
      </p>
    </div>
    <div
      className="selfDate d-flex"
    >
      <p
        className="bold"
      >
        Date:
      </p>
      <p>
        2022-03-07
      </p>
       
    </div>
    <div
      className="selfStats"
    >
      <div
        className="d-flex"
      >
        <p
          className="bold"
        >
          Total Cases:
        </p>
        <p>
          272250
        </p>
         
      </div>
      <div
        className="d-flex"
      >
        <p
          className="bold"
        >
          Total Deaths:
        </p>
        <p>
          3483
        </p>
         
      </div>
      <div
        className="d-flex"
      >
        <p
          className="bold"
        >
          Cases today:
        </p>
        <p>
          40
        </p>
         
      </div>
      <div
        className="d-flex"
      >
        <p
          className="bold"
        >
          Deaths today:
        </p>
        <p>
          1
        </p>
         
      </div>
    </div>
    <div
      className="backButton"
    >
      <i
        aria-hidden="true"
        className="bi bi-arrow-left-circle ck"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Single Country renders 1`] = `
<div
  className="selfCountries d-flex col"
>
  <div
    className="countryContainer d-flex col"
  >
    <div
      className="selfTitle lato bold d-flex"
    >
      <p>
        Albania
      </p>
    </div>
    <div
      className="selfDate d-flex"
    >
      <p
        className="bold"
      >
        Date:
      </p>
      <p>
        2022-03-07
      </p>
       
    </div>
    <div
      className="selfStats"
    >
      <div
        className="d-flex"
      >
        <p
          className="bold"
        >
          Total Cases:
        </p>
        <p>
          272250
        </p>
         
      </div>
      <div
        className="d-flex"
      >
        <p
          className="bold"
        >
          Total Deaths:
        </p>
        <p>
          3483
        </p>
         
      </div>
      <div
        className="d-flex"
      >
        <p
          className="bold"
        >
          Cases today:
        </p>
        <p>
          40
        </p>
         
      </div>
      <div
        className="d-flex"
      >
        <p
          className="bold"
        >
          Deaths today:
        </p>
        <p>
          1
        </p>
         
      </div>
    </div>
    <div
      className="backButton"
    >
      <i
        aria-hidden="true"
        className="bi bi-arrow-left-circle ck"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
